### PR TITLE
[kie-issues#843] Generate reproducible dmndefinitions.json.

### PR DESCRIPTION
**Pull request by baldimir**
- Orders the types written into dmndefinitions.json file.

### CODE REVIEW

File: kie-dmn/kie-dmn-openapi/src/main/java/org/kie/dmn/openapi/impl/DMNOASGeneratorImpl.java [hunk 1]
```diff
@@ -81,7 +82,8 @@ private void prepareSerializaton() {
         ObjectNode tree = JsonUtil.objectNode();
         ObjectNode definitions = JsonUtil.objectNode();
         tree.set("definitions", definitions);
-        for (Entry<DMNType, Schema> kv : schemas.entrySet()) {
+        final List<Entry<DMNType, Schema>> sortedEntries = schemas.entrySet().stream().sorted(Map.Entry.comparingByKey(Comparator.comparing(DMNType::getName))).toList();
+        for (Entry<DMNType, Schema> kv : sortedEntries) {
```
Comment (mariofusco) [inline kie-dmn/kie-dmn-openapi/src/main/java/org/kie/dmn/openapi/impl/DMNOASGeneratorImpl.java:88]: If I understand correctly the purpose of this change is traversing the `schemas` Map in a deterministic order. In this case wouldn't be easier and more readable to turn the `HashMap` into a `TreeMap`, something like:
```
private final Map<DMNType, Schema> schemas = new TreeMap<>(Comparator.comparing(DMNType::getName));
```

Comment (baldimir) [reply kie-dmn/kie-dmn-openapi/src/main/java/org/kie/dmn/openapi/impl/DMNOASGeneratorImpl.java:88]: Thanks, makes sense. Fixed.

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-drools/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-drools -u https://github.com/apache/incubator-kie-drools/pull/5676 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5676/2/display/redirect

**Test results:**
- PASSED: 21922
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/drools/job/main/job/pullrequest_jobs/job/drools-pr/job/PR-5676/2/testReport/org.kie.dmn.openapi/ProcessItemTest/test_together/">org.kie.dmn.openapi.ProcessItemTest.test_together</a></summary>
<br/>Expecting actual not to be empty
</details>

**Review by yesamer — APPROVED** [review]

**Comment by baldimir** [comment]
I will take a look at the test.

**Review by mariofusco — APPROVED** [review]

**Comment by baldimir** [comment]
I reverted the change to a TreeMap as that broke one test. With the sorting I had before, it works. For some reason, some other code, where the Map is sent (or its entries), it relies on it not being sorted or similar (Honestly, such reliance look strange to me, it should not happen). It could take much more time to investigate, so I suggest merging this approach and investigating after the release. What do you think please? @mariofusco @yesamer

**Pull request merged**